automatic/cutepdf/tools/chocolateyInstall.ps1

```diff
@@ -1,16 +1,30 @@
+﻿$ErrorActionPreference = 'Stop'
+
 $checksum = '2D9E07F25358C9D2317BC639AFCDEDDB893D1FCFD43BB66FF372DBA11E169EE1'
 $url = 'http://www.cutepdf.com/download/CuteWriter.exe'
 
 $packageArgs = @{
   packageName   = 'cutepdf'
-  unzipLocation = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
   fileType      = 'exe'
   url           = $url
   silentArgs    = '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-' # Inno Setup Package
   validExitCodes= @(0)
-  softwareName  = 'cutepdf*'
   checksum      = $checksum
   checksumType  = 'sha256'
 }
 
+# Make sure Print Spooler service is up and running
+try {
+  $serviceName = 'Spooler'
+  $spoolerService = Get-WmiObject -Class Win32_Service -Property StartMode,State -Filter "Name='$serviceName'"
+  if ($spoolerService -eq $null) { throw "Service $serviceName was not found" }
+  Write-Host "Print Spooler service state: $($spoolerService.StartMode) / $($spoolerService.State)"
+  if ($spoolerService.StartMode -ne 'Auto' -or $spoolerService.State -ne 'Running') {
+    Set-Service $serviceName -StartupType Automatic -Status Running
```

Comment:
I understand your point of view but if the service is disabled back at the end of the install, cutepdf won't work. So what is the point to install it successfully if it doesn't work in the end?

An other idea would be to cancel the install and warn the user about the reason so that he fixes this but then it means that this package couldn't pass verifier service (since the service is disabled on this environment).

What do you think?